Classes/Command/UpdateChildRelationCommand.php

```diff
@@ -0,0 +1,184 @@
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getQueryBuilderForTable(self::RELATION_INDEX_TABLE);
+
+        $queryBuilder
+            ->insert(self::RELATION_INDEX_TABLE)
+            ->values([
+                'uid_parent' => $record['uid_parent'],
+                'uid_child' => $record['uid_child'],
+                'tablename' => $record['tablename'],
+                'child_parent_id' => $record['child_parent_id'],
+                'child_parent_tablename' => $record['child_parent_tablename'],
+            ])
+            ->execute();
+    }
+
+    /**
+     * Fetch missing child relation.
+     *
+     * @return array
+     */
+    protected function fetchMissingChildRelations(): array
+    {
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getQueryBuilderForTable(self::RELATION_INDEX_TABLE);
+        $queryBuilder->getRestrictions()->removeAll();
+
+        $records = $queryBuilder
+            ->select(
+                'tpdmaparent.uid as uid_parent',
+                'tpdma.uid as uid_child',
+                'tpdmp.uid as child_parent_id'
+            )
+            ->addSelectLiteral(
+                $queryBuilder->expr()->count('*'),
+                '\'tx_pxaproductmanager_domain_model_product\' as child_parent_tablename',
+                '\'tx_pxaproductmanager_domain_model_attributevalue\' as tablename'
+            )
+            ->from(self::PRODUCT_TABLE, 'tpdmp')
+            ->leftJoin(
+                'tpdmp',
+                self::ATTRIBUTEVALUE_TABLE,
+                'tpdma',
+                $queryBuilder->expr()->eq(
+                    'tpdma.product',
+                    $queryBuilder->quoteIdentifier('tpdmp.uid')
+                )
+            )
+            ->leftJoin(
+                'tpdmp',
+                self::ATTRIBUTEVALUE_TABLE,
+                'tpdmaparent',
+                $queryBuilder->expr()->andX(
+                    $queryBuilder->expr()->eq(
+                        'tpdmaparent.product',
+                        $queryBuilder->quoteIdentifier('tpdmp.parent')
+                    ),
+                    $queryBuilder->expr()->eq(
+                        'tpdmaparent.attribute',
+                        $queryBuilder->quoteIdentifier('tpdma.attribute')
+                    )
+                )->__toString()
+            )
+            ->leftJoin(
+                'tpdma',
+                self::RELATION_INDEX_TABLE,
+                'tprii',
+                $queryBuilder->expr()->andX(
+                    $queryBuilder->expr()->eq(
+                        'tprii.uid_child',
+                        $queryBuilder->quoteIdentifier('tpdma.uid')
+                    ),
+                    $queryBuilder->expr()->eq(
+                        'tprii.uid_parent',
+                        $queryBuilder->quoteIdentifier('tpdmaparent.uid')
+                    )
+                )->__toString()
```

Comment:
If you like the above, you'll like this.

```suggestion
                (string)$queryBuilder->expr()->andX(
                    $queryBuilder->expr()->eq(
                        'tprii.uid_child',
                        $queryBuilder->quoteIdentifier('tpdma.uid')
                    ),
                    $queryBuilder->expr()->eq(
                        'tprii.uid_parent',
                        $queryBuilder->quoteIdentifier('tpdmaparent.uid')
                    )
                )
```